pallets/loans/src/rate_model.rs

```diff
@@ -165,19 +185,57 @@ impl CurveModel {
     pub fn new_model(base_rate: Rate) -> CurveModel {
         Self { base_rate }
     }
+}
 
-    /// Check the curve model for sanity
-    pub fn check_model(&self) -> bool {
-        true
+impl InterestRateModel for CurveModel {
+    fn check_model(&self) -> bool {
+        false
     }
 
-    /// Calculates the borrow interest rate of curve model
-    pub fn get_borrow_rate(&self, _utilization: Ratio) -> Option<Rate> {
-        // TODO: Need to implement the curve model
+    fn get_borrow_rate(&self, _: Ratio) -> Option<Rate> {
         None
     }
 }
 
+#[derive(Clone, Copy, Decode, Default, Encode, Eq, PartialEq, RuntimeDebug)]
+pub struct NewCurveNameModel {
+    borrow_rate: Rate,
+    reserve_ratio: Ratio,
+    staking_apy: Ratio,
+}
+
+impl NewCurveNameModel {
+    pub fn new(borrow_rate: Rate, reserve_ratio: Ratio, staking_apy: Ratio) -> Self {
+        Self {
+            borrow_rate,
+            reserve_ratio,
+            staking_apy,
+        }
+    }
+}
+
+impl InterestRateModel for NewCurveNameModel {
+    fn check_model(&self) -> bool {
+        false
```

Comment:
Also couldn't figure out sane values for the struct fields. Any suggestion is welcome

Reply:
See above comment